R/utils-get_code_dependency.R

```diff
@@ -144,6 +145,16 @@ fix_shifted_comments <- function(calls) {
   Filter(nrow, calls)
 }
 
+#' Fixes edge case of `:=` assignment operator being treated as assignment.
+#' @keywords internal
+#' @noRd
+remove_dt_assign <- function(calls) {
+  checkmate::assert_list(calls)
+  lapply(calls, function(call) {
+    call <- subset(call, !(token == "LEFT_ASSIGN" & text == ":="))
```

Comment:
Do you need the assignment here at all?

Reply:
`subset` gives this in the R CMD CHECK notes

```r
remove_dt_assign : <anonymous>: no visible binding for global variable
  ‘token’
remove_dt_assign : <anonymous>: no visible binding for global variable
  ‘text’
Undefined global functions or variables:
  text token
```

I think I'm gonna move back to `object$token` and `object$text`

Reply:
changed to https://github.com/insightsengineering/teal.code/pull/234/commits/9dfc3e7ae2362edd1515663d8147e68f797f1aa9